python/cudf_polars/cudf_polars/experimental/expressions.py

```diff
@@ -0,0 +1,485 @@
+    chunk_aggs: list[Expr]
+    agg_name = agg.name
+    if agg_name == "count":
+        chunk_aggs = [agg]
+        combine_aggs = [rename_agg(agg, "sum")]
+        finalize = None
+    elif agg_name == "mean":
+        sum_agg = rename_agg(agg, "sum")
+        count_agg = rename_agg(agg, "count")
+        chunk_aggs = [sum_agg, count_agg]
+        combine_aggs = [sum_agg, sum_agg]
+        finalize = (agg.dtype, "DIV")
+    elif agg_name == "n_unique":
+        chunk_aggs = [Cast(agg.dtype, agg)]
+        combine_aggs = [rename_agg(agg, "sum")]
+        finalize = None
+        child, graph = shuffle_child(
+            child,
+            agg.children[0],
+            named_expr,
+            child_partition_info,
+            # TODO: Plumb through config options to Shuffle
+            ConfigOptions({}),
+        )
+    else:
+        chunk_aggs = [agg]
+        combine_aggs = [agg]
+        finalize = None
+
+    # Pointwise stage
+    pointwise_keys = []
+    child_name = get_key_name(child)
+    chunk_name = f"chunk-{key_name}"
+    for i in range(input_count):
+        pointwise_keys.append((chunk_name, i))
+        graph[pointwise_keys[-1]] = (
+            evaluate_chunk_multi_agg,
+            (child_name, i),
+            chunk_aggs,
+            expr.children,
+            *[
+                (name, 0) if bcast else (name, i)
+                for name, bcast in zip(expr_child_names, expr_bcast, strict=True)
+            ],
+        )
+
+    # Combine and finalize
+    graph[(key_name, 0)] = (
+        combine_chunks_multi_agg,
+        pointwise_keys,
+        combine_aggs,
+        finalize,
+        named_expr.name,
+    )
+
+    return graph
+
+
+def make_pointwise_graph(
+    named_expr: NamedExpr,
+    expr_partition_counts: MutableMapping[Expr, int],
+    child: IR,
+) -> MutableMapping[Any, Any]:
+    """Build simple pointwise FusedExpr graph."""
+    expr = named_expr.value
+    child_name = get_key_name(child)
+    assert isinstance(expr, FusedExpr)
+    key_name = get_key_name(expr)
+    expr_child_names = [get_key_name(c) for c in expr.children]
+    expr_bcast = [expr_partition_counts[c] == 1 for c in expr.children]
+    count = expr_partition_counts[expr]
+    sub_expr = named_expr.reconstruct(expr.sub_expr)
+    return {
+        (key_name, i): (
+            evaluate_chunk,
+            (child_name, i),
+            sub_expr,
+            expr.children,
+            *[
+                (name, 0) if bcast else (name, i)
+                for name, bcast in zip(expr_child_names, expr_bcast, strict=True)
+            ],
+        )
+        for i in range(count)
+    }
+
+
+def make_fusedexpr_graph(
+    named_expr: NamedExpr,
+    expr_partition_counts: MutableMapping[Expr, int],
+    child: IR,
+    child_partition_info: PartitionInfo,
+) -> MutableMapping[Any, Any]:
+    """Build task graph for a FusedExpr node."""
+    expr = named_expr.value
+    assert isinstance(expr, FusedExpr)
```

Comment:
Are these assertions to pacify the type-checker, or because you're worried that these aren't going to be `FusedExpr`s?

Reply:
I think I moved this one by accident, but yes - This is to pacify type checking.